### PR TITLE
Docs: Switch to using MyST Markdown

### LINKED ISSUE #924: 📚Docs: Switch to using MyST Markdown
We should convert our documentation into MyST Markdown:

https://myst-parser.readthedocs.io/en/latest/index.html

Using RST is a bit tricky since you always have to relearn the syntax, while Markdown is used _everywhere_. We can also just copy-paste text from our (sometimes quite extensive) GitHub of Slack discussions without having to adapt the syntax.

This also would allow us to start integrating Jupyter notebooks using MyST-NB and `jupytext`, making it also possible to write documentation that can be tested.

### PR DESCRIPTION
**Pull request by mbercx**
Fixes #924 

Convert the entire documentation in MyST Markdown, which is easier to use due to its
similarity with the ubiquitous Markdown and cleaner syntax.

Moving to MyST Markdown also makes it easier/more consistent to use [MyST-NB](https://myst-nb.readthedocs.io/en/latest/index.html)
and eventually move to fully executable documentation.

Some remaining RestructuredText elements:

* The `autodoc` generation of the API. There is also 
   https://sphinx-autodoc2.readthedocs.io/en/latest/index.html. I remember there
   was a reason I didn't use it, but I don't remember why. ^^
* Automated generation of CLI documantation.
* `aiida.sphinxext` items

### CODE REVIEW

**Comment by sphuber** [comment]
Thanks @mbercx . I'm fine with the move in principle. Just noticed that the styling is quite different in the new build. Is this because some styling rules are not migrated properly yet, or because one of the updated dependencies changes the style? Did `sphinx-book-theme` change the appearance that much? I think I prefer the current style personally

**Comment by mbercx** [comment]
I think the styling changes are mainly due to the bump in the `sphinx-book-theme` indeed, which was necessary. We can change the defaults for sure, though, just let me know which changes you're not a fan of and I can look into it.

**Review by sphuber — CHANGES_REQUESTED** [review]
Thanks @mbercx . Let's not get into bikeshedding on the styling. It's more important to get this in so it becomes easier to start adding additional docs. I have one open question though.

File: rst2myst.yaml
[the remark this answers is not in the document]
Comment (mbercx) [reply rst2myst.yaml:1]: Well spotted! That one snuck in accidentally indeed. Removed 👍

**Review by sphuber — APPROVED** [review]

**Pull request merged**